src/ast/passes/codegen_llvm.cpp

```diff
@@ -2753,6 +2753,27 @@ ScopedExpr CodegenLLVM::visit(ArrayAccess &arr)
   auto scoped_expr = visit(arr.expr);
   auto scoped_index = visit(arr.indexpr);
 
+  if (type.IsArrayTy()) {
+    llvm::Function *parent = b_.GetInsertBlock()->getParent();
+    BasicBlock *is_oob = BasicBlock::Create(module_->getContext(),
+                                            "is_oob",
+                                            parent);
+    BasicBlock *merge = BasicBlock::Create(module_->getContext(),
+                                           "oob_merge",
+                                           parent);
+
+    Value *cond = b_.CreateICmpUGT(
+        b_.CreateIntCast(scoped_index.value(), b_.getInt64Ty(), false),
+        b_.getInt64(type.GetNumElements() - 1),
```

Comment:
What is `GetNumElements` for a variable-sized array? This should have a test, and maybe you just don't do this check?

Reply:
And it seems variable-sized array `GetNumElements` returns `0`.